WordPress/pattern-directory
# Add a setting to change the flag threshold

Adds a setting field to the existing Block Pattern Settings page to adjust the number of times a pattern can be reported before it is automatically unpublished.

This also moves the code that creates the Block Pattern Settings page from the pattern-creator plugin over to the pattern-directory plugin, as discussed [here](https://github.com/WordPress/pattern-directory/issues/425#issuecomment-1069766718).

Fixes #425 

### Screenshot

![pattern-settings](https://user-images.githubusercontent.com/916023/158870168-34d87dd1-db78-4ea8-85f3-701c0a8fdf33.jpg)

### How to test the changes in this Pull Request:

1. Change the flag threshold setting to `1`. View a published pattern on the front end and report it. The pattern should now be set to "pending".
2. Change the flag threshold back to `5`. Now if you report a pattern, it won't be set to "pending" (until 4 other users also report it).
